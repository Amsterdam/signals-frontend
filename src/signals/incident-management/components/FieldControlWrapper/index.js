import React from 'react';
import PropTypes from 'prop-types';

import { FieldControl } from 'react-reactive-form';

const FieldControlWrapper = ({ name, control, render, ...props }) => (
  <div>
    <FieldControl
      name={name}
      control={control}
      render={render({
        ...props,
        name
      })}
    />
  </div>);

FieldControlWrapper.propTypes = {
  name: PropTypes.string.isRequired,
  control: PropTypes.object.isRequired,
  render: PropTypes.func.isRequired,
};

export default FieldControlWrapper;
