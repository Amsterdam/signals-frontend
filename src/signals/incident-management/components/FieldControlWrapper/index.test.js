import React from 'react';
import { shallow } from 'enzyme';

import { FormControl } from 'react-reactive-form';
import FieldControlWrapper from './index';

describe('FieldControlWrapper', () => {
  it('should render correctly', () => {
    const props = { name: 'name', control: new FormControl(), render: jest.fn() };
    const wrapper = shallow(<FieldControlWrapper {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
