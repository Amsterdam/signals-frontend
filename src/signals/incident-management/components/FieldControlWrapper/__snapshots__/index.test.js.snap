// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldControlWrapper should render correctly 1`] = `
<div>
  <FieldControl
    control={
      FormControl {
        "_onDisabledChange": Array [],
        "_pendingChange": true,
        "_pendingDirty": false,
        "_pendingTouched": false,
        "_pendingValue": undefined,
        "asyncValidator": null,
        "errors": null,
        "formState": undefined,
        "get": [Function],
        "getError": [Function],
        "handler": [Function],
        "hasError": [Function],
        "meta": Object {},
        "onBlur": [Function],
        "onBlurChanges": Observable {
          "observers": Array [],
        },
        "onChange": [Function],
        "onValueChanges": Observable {
          "observers": Array [],
        },
        "patchValue": [Function],
        "pristine": true,
        "reset": [Function],
        "setValue": [Function],
        "stateChanges": Observable {
          "observers": Array [],
        },
        "status": "VALID",
        "statusChanges": Observable {
          "observers": Array [],
        },
        "submitted": false,
        "touched": false,
        "validator": null,
        "validatorsOrOpts": undefined,
        "value": undefined,
        "valueChanges": Observable {
          "observers": Array [],
        },
      }
    }
    name="name"
    strict={true}
  />
</div>
`;
